AttitudeManager/ControlAlgorithms/Inc/AM_FixedControl.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * AM_LevelQuadControl.hpp
+ *
+ * Attitude Manager Level Mode Quad Controller Algorithm
+ *
+ * Created on: Oct 22, 2022
+ * Author(s): Aidan Bowers
+ */
+#ifndef ZPSW3_AM_FIXED_CONTROL_HPP
+#define ZPSW3_AM_FIXED_CONTROL_HPP
+
+#include "AM_ControlInterface.hpp"
+
+namespace AM {
+
+class FixedControl : public ControlInterface {
+   public:
+    FixedControl(ActuatorConfig &engine, ActuatorConfig &leftAileron,
+                 ActuatorConfig &rightAileron, ActuatorConfig &rudder,
+                 ActuatorConfig &elevator)
+        : ControlInterface(NumActuatorIdx),
+          configs{engine, leftAileron, rightAileron, rudder, elevator} {};
+
+    std::vector<ActuatorOutput> runControlsAlgo(
+        const AttitudeManagerInput &instructions) const override;
+
+    void updatePid() override { return; }
+
+   private:
+    enum ActuatorIdx {
+        Engine = 0,
+        LeftAileron,
+        RightAileron,
+        Rudder,
+        Elevator,
```

Comment:
There will be different control algorithms for different aircraft configurations (fixed wing/quadcopter/helicopter). Each of those frames will then take a state mix that determines how each actuator impacts the state of the aircraft when...actuated in a positive direction.

I believe SM will get to choose which controls algorithms to run? not sure about that tbh.

Reply:
> Oh wait, since this is in the controlAlgorithm folder that means there will be different controls algorithms based on different frames? And the CMake will be configured to choose which one based on some flag or something?

You got it, different control algorithm for different air frames. I don't think we will be choosing what ones to use with a compiler flag though (especially since we may want more than 1 and to transition between them (e.g. fixed wing to quad on our planned air fame)